scripts/keyless/keylessDeploy.ts

```diff
@@ -0,0 +1,293 @@
+    gasPrice: 100 * 10 ** 9,
+    gasLimit: 5000000,
+    to: commonFactory,
+    value: 0,
+    nonce: 0,
+    data: data,
+  };
+  const customSignature = ethers.utils.joinSignature(customSigInfo);
+  const serializedTestTx = ethers.utils.serializeTransaction(testTx);
+
+  const addr = ethers.utils.recoverAddress(
+    ethers.utils.arrayify(ethers.utils.keccak256(serializedTestTx)),
+    customSignature,
+  );
+
+  const signedSerializedTestTx = ethers.utils.serializeTransaction(testTx, customSigInfo);
+
+  return {
+    addr,
+    signedSerializedTestTx,
+  };
+}
+
+export async function computeDeploymentAddress(bytecode: any, args: any) {
+  const bytecodeHash = hardhatEthers.utils.id(bytecode);
+  const salt = `tw.${bytecodeHash}`;
+  const saltHash = hardhatEthers.utils.id(salt);
+
+  const deployData = hardhatEthers.utils.solidityPack(["bytes32", "bytes", "bytes"], [saltHash, bytecode, args]);
+
+  const initBytecode = hardhatEthers.utils.solidityPack(["bytes", "bytes"], [bytecode, args]);
+  const deployInfoPacked = hardhatEthers.utils.solidityPack(
+    ["bytes1", "address", "bytes32", "bytes32"],
+    ["0xff", commonFactory, saltHash, hardhatEthers.utils.solidityKeccak256(["bytes"], [initBytecode])],
+  );
+  const addr = hardhatEthers.utils.solidityKeccak256(["bytes"], [deployInfoPacked]);
+
+  return { predictedAddress: `0x${addr.slice(26)}`, deployData };
+}
+
+async function deployInfraKeyless() {
+  const [signer] = await hardhatEthers.getSigners();
+
+  // create2 factory
+  await deployCommonFactory(signer);
+
+  for (let txInfo of Object.values(infraContracts) as InfraTxInfo[]) {
+    const code = await hardhatEthers.provider.getCode(txInfo.predictedAddress);
+    if (code === "0x") {
+      let fundAddr = {
+        to: txInfo.from,
+        value: hardhatEthers.utils.parseEther("1"),
+      };
+      await signer.sendTransaction(fundAddr);
+
+      await (await hardhatEthers.provider.sendTransaction(txInfo.tx)).wait();
+    }
+  }
+}
+
+async function deployInfraWithSigner() {
+  const [signer] = await hardhatEthers.getSigners();
+
+  // create2 factory
+  await deployCommonFactory(signer);
+
+  for (let txInfo of Object.values(infraContracts) as InfraTxInfo[]) {
+    const block = await hardhatEthers.provider.getBlock("latest");
+    const baseBlockFee =
+      block && block.baseFeePerGas ? block.baseFeePerGas : hardhatEthers.utils.parseUnits("1", "gwei");
+    const feeData = await hardhatEthers.provider.getFeeData();
+    const baseMaxFeePerGas = baseBlockFee.mul(2);
+    const maxPriorityFeePerGas = BigNumber.from(feeData.maxPriorityFeePerGas);
+
+    // get init bytecode
+    const deployData = await hardhatEthers.provider.getCode(txInfo.deployData);
+
+    const tx = {
+      from: signer.address,
+      to: commonFactory,
+      value: 0,
+      nonce: await signer.getTransactionCount("latest"),
+      maxPriorityFeePerGas: maxPriorityFeePerGas,
+      maxFeePerGas: baseMaxFeePerGas.add(maxPriorityFeePerGas),
```

Comment:
i dont think you actually have to pass the gas fee here right? the provider should handle it for you i believe